File: school/summer2013/openmp/c/06_local_reduction.c
#include <stdio.h>
#include <omp.h>

int main()
{
	long i, sum=0, local_sum;

#pragma omp parallel private(i, local_sum)
{
	local_sum = 0;
	#pragma omp for
	for(i=1; i<=N; i++) 
		local_sum += i;

	#pragma omp atomic
	sum += local_sum;
}

	printf("sum = %ld\n", sum);
}

File: school/summer2013/openmp/fortran/06_local_reduction.f90
program local_reduction
	implicit none
    integer(8) :: i, sum=0, local_sum

!$omp parallel private(local_sum)
    local_sum = 0
    !$omp do
    do i=1, N
        local_sum = local_sum + i
    end do
    !$omp atomic
    sum = sum + local_sum
!$omp end parallel

    print *, 'sum =', sum
end
